compiler/tests/22_pex_func_test.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python2.7
+"""
+Run a regresion test on various srams
+"""
+
+import unittest
+from testutils import header,openram_test
+import sys,os
+sys.path.append(os.path.join(sys.path[0],".."))
+import globals
+from globals import OPTS
+import debug
+
+class sram_func_test(openram_test):
+
+    def runTest(self):
+        globals.init_openram("config_20_{0}".format(OPTS.tech_name))
+        global verify
+        import verify
+
+        #Initialize options
+        OPTS.check_lvsdrc = False
+        OPTS.use_pex = False
+        OPTS.spice_name="" # Unset to use any simulator
+        OPTS.analytical_delay = False
+
+        # This is a hack to reload the characterizer __init__ with the spice version
+        import characterizer
+        reload(characterizer)
+        from characterizer import delay
+        if not OPTS.spice_exe:
+            debug.error("Could not find {} simulator.".format(OPTS.spice_name),-1)
+
+        import sram
+
+        #create sram test bank
+        debug.info(1, "Testing timing for sample 1bit, 16words SRAM with 1 bank")
+        s = sram.sram(word_size=2,
+                      num_words=16,
+                      num_banks=1,
+                      name="pex_sram_test")
+
+        #important
+        OPTS.check_lvsdrc = True
+        OPTS.use_pex = True
+
+        # trimming the netlist doesn't make much time difference
+        # OPTS.trim_netlist = False
+
+        #Write a temporary spice file for analysis
+        tempspice = OPTS.openram_temp + "temp.sp"
+        tempgds = OPTS.openram_temp + "temp.gds"
+        s.sp_write(tempspice)
+        s.gds_write(tempgds)
+
+        #Run and debug pex test. This goes to verify folder and run_pex function
+        self.assertFalse(verify.run_pex(s.name, tempgds, tempspice, output=None))
```

Comment:
Is there any checking of what the final result should be? This is successful if the test runs, but no result is checked.

Reply:
I can look at the output file, but I don't have much of an idea of what to compare it to. Is there some sort of standard we can measure the output against?

Reply:
I'm not sure. For the other delay tests, we just ensure that the delay matches a correct value that we measured with spice by hand. This requires constant updating every time something changes though. Even if we allow a 10% or so difference. So, I'm not sure. However, just checking that it ran is not enough...

Reply:
Would it be enough to compare the delay values to the results for the unextracted netlist (I was thinking the numbers used in "21_hspice_delay_test". If the numbers are greater from the extracted delay than those used in the unextracted test, it may work. The delay would be larger from including parasitics, but it would not be accurate. I remember you said in an email earlier that it didn't have to check how well it works, just that it works. I understand that this would not be entirely accurate, but it would be closer than nothing.

Am I totally off base with this idea?